processor/memorylimiter/memorylimiter.go

```diff
@@ -138,12 +134,26 @@ func getMemUsageChecker(cfg *Config, logger *zap.Logger) (*memUsageChecker, erro
 		return nil, fmt.Errorf("failed to get total memory, use fixed memory settings (limit_mib): %w", err)
 	}
 	logger.Info("Using percentage memory limiter",
-		zap.Uint64("total_memory", totalMemory),
+		zap.Uint64("total_memory_mib", totalMemory/mibBytes),
 		zap.Uint32("limit_percentage", cfg.MemoryLimitPercentage),
 		zap.Uint32("spike_limit_percentage", cfg.MemorySpikePercentage))
 	return newPercentageMemUsageChecker(totalMemory, uint64(cfg.MemoryLimitPercentage), uint64(cfg.MemorySpikePercentage))
 }
 
+func (ml *memoryLimiter) start(_ context.Context, host component.Host) error {
+	extensions := host.GetExtensions()
+	for id, extension := range extensions {
+		if id == config.NewID("memory_ballast") {
+			if ext, ok := extension.(*ballastextension.MemoryBallast); ok {
+				ml.ballastSize = ext.GetBallastSize()
```

Comment:
Break after found?
Would be also nice to log what is happening so that it is not a surprise for the user.